### ISSUE TITLE
[0.9.4.2] should not throw error message when trying to use DERIVATIVE with WHERE time() clause

### ISSUE THREAD

**Issue by erdii** [influxdata/influxdb#4667]
When querying without a WHERE i get tons of results as expected:
```
> SELECT derivative(recv) FROM rsmq_monitor..queue1
....
2015-11-05T09:30:39Z	43
2015-11-05T09:30:40Z	42
2015-11-05T09:30:41Z	38
2015-11-05T09:30:42Z	41
2015-11-05T09:30:43Z	37
2015-11-05T09:30:44Z	37
2015-11-05T09:30:45Z	38
2015-11-05T09:30:46Z	41
2015-11-05T09:30:47Z	38
2015-11-05T09:30:48Z	41
2015-11-05T09:30:49Z	50
2015-11-05T09:30:50Z	44
2015-11-05T09:30:51Z	39
2015-11-05T09:30:52Z	36
...
```

however when i try to limit the same query with `WHERE time > now() - 10m`, I get an error:
```
> SELECT derivative(recv) FROM rsmq_monitor..queue1 WHERE time > now() - 10m
ERR: error parsing query: aggregate function required inside the call to derivative
```

**Comment by beckettsean** [comment]
This is expected behavior. When using a `GROUP BY time()` the `DERIVATIVE` function calculates the rate of change between the `GROUP BY` intervals. That means each interval must have a single value, not many. An aggregation function is required so that each interval has one and only one value.

**Comment by beckettsean** [comment]
@rkuchan something to note for the Functions page

**Comment by beckettsean** [comment]
I think the error should mention that the aggregate function is only required when `DERIVATIVE` is used in conjunction with `GROUP BY`

**Comment by otoolep** [comment]
I don't see him use GROUP BY.

**Comment by otoolep** [comment]
OK, I see `GROUP BY` in the comment description, but I don't see it in the queries shown in the ticket.

**Comment by beckettsean** [comment]
My mistake, I misread the query that causes the error.

This looks like a fault with the parser where it is incorrectly throwing the error.

Thanks for the report, @erdii and my apologies for misreading your query.

**Comment by beckettsean** [comment]
@erdii what version of InfluxDB are you running?

**Comment by erdii** [comment]
No problem dude!

I'm using InfluxDB 0.9.4.2 on OS X 10.11.1 installed via homebrew

**Comment by jsternberg** [comment]
This appears to be fixed in the latest 0.12.
